unlock-app/src/hooks/useMultipleRecipient.ts

```diff
@@ -0,0 +1,198 @@
+  const lockByAddress = lockAddress ? locks?.[lockAddress] : {}
+  const normalizeRecipients = () => {
+    if (!lockAddress) return
+
+    /*
+      we need to exclude user metadata that already already exists,
+      by creating a new array without items in "ignoreRecipients"
+     */
+    const listWithoutExcluded = recipientsList().filter(
+      ({ resolvedAddress }) => {
+        const exludedAddressList = (ignoreRecipients ?? []).map(
+          ({ userAddress }) => userAddress
+        )
+        const isAddressExcluded =
+          (exludedAddressList?.length > 0 &&
+            exludedAddressList.includes(resolvedAddress)) ||
+          exludedAddressList.length === 0
+        return isAddressExcluded
+      }
+    )
+
+    const payload: RecipientPayload = {
+      users: listWithoutExcluded.map(({ resolvedAddress, metadata = {} }) => {
+        return {
+          userAddress: resolvedAddress,
+          metadata,
+          lockAddress,
+        }
+      }),
+    }
+    return payload
+  }
+
+  const clear = () => {
+    setRecipients(new Set([]))
+  }
+
+  useEffect(() => {
+    const activeMultiple = +maxRecipients > 1
+    setHasMultipleRecipients(activeMultiple)
+  }, [])
+
+  const recipientsList = (): RecipientItem[] => {
+    return Array.from(recipients)
+  }
+
+  const canAddUser = () => {
+    return recipients.size < maxRecipients
+  }
+
+  const getAddressAndValidation = async (recipient: string) => {
+    const address = await getAddressForName(recipient)
+    return {
+      valid: address?.length > 0 ?? false,
+      address,
+    }
+  }
+
+  const submitBulkRecipients = async () => {
+    if (!lockByAddress?.network) return
+    const url = `${config.services.storage.host}/v2/api/metadata/${lockByAddress?.network}/users`
+    const opts = {
+      method: 'POST',
+      body: JSON.stringify(normalizeRecipients()),
+      headers: {
+        'content-type': 'application/json',
+      },
+    }
+    const response = await fetch(url, opts)
+    return response
+  }
+
+  const submit = async (): Promise<boolean> => {
+    if (retryCount < MAX_RETRY_COUNT) {
+      const res = await submitBulkRecipients()
+      if (res) {
+        const result: any = await res.json()
+        const errorCodes = [500, 401, 404]
+        const successCodes = [200, 201]
+        setIgnoreRecipients([])
+
+        if (res.status === 409) {
+          const addressesWithMetadata = result?.users?.map(
+            ({ userAddress }: any) => userAddress
+          )
+          setIgnoreRecipients(addressesWithMetadata)
+          setRetryCount(retryCount + 1)
+          ToastHelper.error(`${result?.message}` ?? 'Ops, something went wrong')
+        }
+
+        if (errorCodes.includes(res.status)) {
+          ToastHelper.error(result?.message ?? 'Ops, something went wrong')
+          setRetryCount(retryCount + 1)
+        }
+
+        if (successCodes.includes(res.status)) {
+          ToastHelper.success('Success')
```

Comment:
This message is vague. Try making it more concrete `Successfully purchased keys for xyz` or something like that.